main/views.py

```
@@ -34,11 +36,52 @@ def add_employee(request):
+
+
+def view_xml_by_id(request, pk):
+    try:
+        productData = get_object_or_404(Product, pk=pk)
+        xml_data = serializers.serialize("xml", [productData])
+        return HttpResponse(xml_data, content_type="application/xml")
+    except Product.DoesNotExist:
+        return HttpResponse("Product not found", status=404)
+        
+    
+def view_json_by_id(request, pk):
```

Ruff (0.12.2)

Unused function argument: `request`

(ARG001)